### PR TITLE
tables: Re-enable required-column warning message

### PR DESCRIPTION
**Pull request by theopolis**
This is an alternative to #6014.

Instead of returning an error from xBestIndex we return an impossible cost.

### CODE REVIEW

File: osquery/sql/virtual_table.cpp [hunk 1]
```diff
@@ -778,26 +782,33 @@ static int xBestIndex(sqlite3_vtab* tab, sqlite3_index_info* pIdxInfo) {
...
+        auto real_column_index = pVtab->content->aliases[column_name];
+        bit = real_column_index < 63 ? real_column_index : 63U;
+        colsUsedBitset.set(bit);
+        column_name = std::get<0>(columns[real_column_index]);
+      }
+      colsUsed.insert(column_name);
+
+      const auto& options = std::get<2>(columns[i]);
+      if (options & ColumnOptions::REQUIRED) {
+        numRequiredColumns++;
```
Comment (directionless) [inline osquery/sql/virtual_table.cpp]: It looks like it's always been an int, but I wonder why it's not a bool

File: docs/wiki/deployment/debugging.md [hunk 2]
```diff
@@ -125,6 +125,11 @@ While not expected, the backing store may be corrupted by problems with the file
 
 If your `--database_path` is `/var/osquery/osquery.db` then the backup is `/var/osquery/osquery.db.backup`. The database is always a folder and the backup location is the suffix ".backup" appended.
 
+### No query solution errors
+
+You may see `no query solution` errors in status logs, as the result of an `osqueryi` query, etc.
+Tables that have a required column are documented on [the schema page](https://osquery.io/schema/).
```
Comment (directionless) [inline docs/wiki/deployment/debugging.md]: cut and paste error?
```suggestion
This typically means a REQUIRED columns must be present in the WHERE clause. For example, the `file` table needs either a `directory` or `path`.
See the [the schema page](https://osquery.io/schema/) for table required column documentation.
```

cut and paste error? (and rewording a sentence)

Comment (theopolis) [reply docs/wiki/deployment/debugging.md]: Actually, missing REQUIRED columns will not give this error anymore. :)

Comment (theopolis) [reply docs/wiki/deployment/debugging.md]: I might just remove this section.

**Comment by theopolis** [comment]
Thanks for the review, applied the feedback.

File: docs/wiki/deployment/debugging.md (hunk 2 above)
Comment (directionless) [reply docs/wiki/deployment/debugging.md]: Removal seems fine.

**Comment by theopolis** [comment]
```
./osquery/osqueryd -S "select * from file" --disable_extensions   
W1116 17:53:37.521584 175668672 virtual_table.cpp:963] Table file was queried without a required column in the WHERE clause
W1116 17:53:37.522099 175668672 virtual_table.cpp:974] Please see the table documentation: https://osquery.io/schema/#file
Error: constraint failed
```

**Comment by Smjert** [comment]
> ```
> ./osquery/osqueryd -S "select * from file" --disable_extensions   
> W1116 17:53:37.521584 175668672 virtual_table.cpp:963] Table file was queried without a required column in the WHERE clause
> W1116 17:53:37.522099 175668672 virtual_table.cpp:974] Please see the table documentation: https://osquery.io/schema/#file
> Error: constraint failed
> ```

Nice!
Returning a `SQLITE_CONSTRAINT` from xFilter instead of `SQLITE_OK` seems also to fix the issue of multiple xFilter called (and so multiple warning messages) when you query a table that has a column `index` which is not `required`, another column which is and the query uses a constraint on the `index` column, but doesn't on the other `required` column.

File: osquery/sql/tests/sqlite_util_tests.cpp [hunk 3]
```diff
@@ -214,17 +214,6 @@ TEST_F(SQLiteUtilTests, test_get_query_columns) {
...
-  std::string query =
-      "SELECT * FROM time, osquery_info, (SELECT * FROM file) ff GROUP BY pid";
-  std::vector<std::string> tables;
-  auto status = getQueryTables(query, tables);
-  EXPECT_TRUE(status.ok());
-
-  std::vector<std::string> expected = {};
-  EXPECT_EQ(expected, tables);
-}
-
```
Comment (Smjert) [inline osquery/sql/tests/sqlite_util_tests.cpp]: Curiosity, why was this removed?

Comment (theopolis) [reply osquery/sql/tests/sqlite_util_tests.cpp]: Good question, I’ll revisit.

File: osquery/sql/tests/sqlite_util_tests.cpp [hunk 4]
```diff
@@ -215,14 +215,9 @@ TEST_F(SQLiteUtilTests, test_get_query_columns) {
...
-      "SELECT * FROM time, osquery_info, (SELECT * FROM file) ff GROUP BY pid";
-  std::vector<std::string> tables;
-  auto status = getQueryTables(query, tables);
-  EXPECT_TRUE(status.ok());
-
-  std::vector<std::string> expected = {};
-  EXPECT_EQ(expected, tables);
+  auto dbc = getTestDBC();
+  QueryDataTyped results;
+  EXPECT_FALSE(queryInternal("SELECT * FROM file", results, dbc).ok());
```
Comment (Smjert) [inline osquery/sql/tests/sqlite_util_tests.cpp]: nit: is there any error message/code we can check to be sure that the failure is caused by a failed constraint?

Comment (Smjert) [reply osquery/sql/tests/sqlite_util_tests.cpp]: I quickly tested and indeed it seems that from here https://github.com/osquery/osquery/blob/8bab38e30d7ac73738c40860272b2268a07fe72c/osquery/sql/sqlite_util.cpp#L605 we get the "constraint failed" message inside the Status.

**Comment by verntx** [comment]
I believe this change invalidates the comment at [line](https://github.com/osquery/osquery/blob/master/osquery/sql/virtual_table.cpp#L770).
